### PR TITLE
Define default provisioning params on plans

### PR DESCRIPTION
**Pull request by carolynvs**
This allows an operator to define a plan and provide a default set of provisioning parameters. I am doing this incrementally so that it's not a giant PR. There will be follow-up PRs to add: `svcat create plan --from`, defaults on classes, etc. The set of milestones is laid out in the [proposal](https://github.com/carolynvs/service-catalog/blob/default-service-plan-proposal/docs/proposals/default-service-plans.md#default-parameters).

TODO:
- [x] Add feature gate
- [x] Documentation

Doc preview available at https://deploy-preview-2282--svc-cat.netlify.com/docs/service-plan-defaults/

### CODE REVIEW

File: docs/service-plan-defaults.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```console
+$ svcat provision mydb --class mysql --plan custom-mysql
+  Name:        mydb
+  Namespace:   default
+  Status:
+  Class:       mysql
+  Plan:        custom-mysql
+
+Parameters:
+  No parameters defined
```
Comment (jberkhahn) [inline docs/service-plan-defaults.md]: this was confusing a bit at first glance. I guess this is just the stuff provided from the svcat command the user entered?

File: Gopkg.lock [hunk 2]
```diff
@@ -1184,6 +1189,7 @@
     "pkg/util"
   ]
   revision = "6702109cc68eb6fe6350b83e14407c8d7309fd1a"
+  version = "kubernetes-1.11.0"
```
Comment (jboyd01) [inline Gopkg.lock]: There are some changes in here I wasn't expecting, but I guess that is just due to a newer point release of Dep you are using?  ie there are some differences in version and branch elements.

File: docs/service-plan-defaults.md (hunk 1 above)
Comment (carolynvs) [reply docs/service-plan-defaults.md]: Correct, it is whatever they first entered in. I agree it's confusing and would like to do a follow-up PR for the svcat UX. This is a pretty big PR as-is, so I tried to limit it to just the server side changes and documentation.

File: Gopkg.lock (hunk 2 above)
Comment (carolynvs) [reply Gopkg.lock]: This has to do with the information available in git at the time we last edited the lock file. When someone updated us to 1.11, they pointed at the branch, not the tag I'm assuming.

This won't change the commit used in our vendor directory, just tweaks the breadcrumb trail on how to find that SHA (i.e. prefer the tag over a moving branch).

**Comment by jeremyrickard** [comment]
I gave this a spin on my AKS cluster and it looks good! 

/lgtm

**Pull request merged**